source/server/listener_manager_impl.cc

```diff
@@ -262,12 +277,27 @@ ListenerImpl::ListenerImpl(const envoy::api::v2::Listener& config, const std::st
     }
   }
 
-  if (config.filter_chains().empty()) {
-    // If we got here, this is a tcp listener, so ensure there is a filter chain specified
+  if (config.filter_chains().empty() &&
+      (socket_type_ == Network::Address::SocketType::Stream || is_quic)) {
     throw EnvoyException(fmt::format("error adding listener '{}': no filter chains specified",
                                      address_->asString()));
   }
+  Server::Configuration::TransportSocketFactoryContextImpl factory_context(
+      parent_.server_.admin(), parent_.server_.sslContextManager(), *listener_scope_,
+      parent_.server_.clusterManager(), parent_.server_.localInfo(), parent_.server_.dispatcher(),
+      parent_.server_.random(), parent_.server_.stats(), parent_.server_.singletonManager(),
+      parent_.server_.threadLocal(), validation_visitor, parent_.server_.api());
+  factory_context.setInitManager(initManager());
+  ListenerFilterChainFactoryBuilderPtr builder =
```

Comment:
if I'm not lost in braces, we'll now do work for raw UDP that we didn't used to.  Is that correct, and is it intentional?